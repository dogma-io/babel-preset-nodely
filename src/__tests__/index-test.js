import preset from '../'

describe('babel-preset-nodely', () => {
  let api

  beforeEach(() => {
    api = {
      cache: {
        never: jest.fn(),
      },
    }
  })

  it('should export a function', () => {
    expect(preset).toBeInstanceOf(Function)
  })

  describe('when no options provided', () => {
    let result

    beforeEach(() => {
      result = preset(api)
    })

    it('should never cache config', () => {
      expect(api.cache.never).toHaveBeenCalled()
    })

    it('should provide expected config', () => {
      expect(result).toMatchSnapshot()
    })
  })

  describe('when env is node', () => {
    let result

    beforeEach(() => {
      result = preset(api, {env: 'node'})
    })

    it('should never cache config', () => {
      expect(api.cache.never).toHaveBeenCalled()
    })

    it('should provide expected config', () => {
      expect(result).toMatchSnapshot()
    })
  })

  describe('when env is web', () => {
    let result

    beforeEach(() => {
      result = preset(api, {env: 'web'})
    })

    it('should never cache config', () => {
      expect(api.cache.never).toHaveBeenCalled()
    })

    it('should provide expected config', () => {
      expect(result).toMatchSnapshot()
    })
  })
})
