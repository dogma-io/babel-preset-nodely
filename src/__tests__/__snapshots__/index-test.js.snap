// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-nodely when env is node should provide expected config 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-syntax-object-rest-spread",
  ],
  "presets": Array [
    Array [
      "@babel/env",
      Object {
        "targets": Object {
          "node": "6",
        },
      },
    ],
    "@babel/flow",
    "@babel/react",
  ],
}
`;

exports[`babel-preset-nodely when env is web should provide expected config 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-syntax-object-rest-spread",
  ],
  "presets": Array [
    Array [
      "@babel/env",
      Object {
        "targets": Object {
          "browsers": Array [
            "last 2 versions",
            "ie 10",
          ],
        },
      },
    ],
    "@babel/flow",
    "@babel/react",
  ],
}
`;

exports[`babel-preset-nodely when no options provided should provide expected config 1`] = `
Object {
  "plugins": Array [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-syntax-object-rest-spread",
  ],
  "presets": Array [
    Array [
      "@babel/env",
      Object {
        "targets": Object {
          "node": "6",
        },
      },
    ],
    "@babel/flow",
    "@babel/react",
  ],
}
`;
